IDLE_DWELL/Find_Idle_Dwells.py

```diff
@@ -0,0 +1,256 @@
+                      "first_AA00_flag": False,
+                      "radmon_status": "UNK"}
+
+# Variable for the stop science command string.
+stop_science_command = "AA00000000"
+
+# If the review load is a full load but only the vehicle portion is to be
+# executed, do nothing as LR was instructed to ignore ACISPKT commands
+if (dwell_args.vo != "VOR") and \
+   (dwell_args.vo != "VOB"):
+
+    # Capture the path to the OFLS directory
+    load_week_path = dwell_args.load_week_path
+    
+    # Find out the status at the start of this load by reading the load's
+    # ACIS-Continuity.txt file to see if this is a Normal load or  other
+    #        (TOO, SCS-107, FULL STOP).
+    # If Normal get the values in the continuity load's ACIS-History.txt file.
+    # If not Normal, get the values in the review load's ACIS-History_edit.txt file.
+
+    # As an extra bonus, you will know whether you are in the Radzone or not
+    # because the ACIS-History[_edit].txt file gives you the RADMON status
+
+    # Read the ACIS-Continuity.txt file
+    continuity_dict = oflsfu.Read_ACIS_Continuity_file(load_week_path)
+
+    # If this load is a Normal load then use the ACIS-History.txt file located
+    # In the continuity file ofls directory
+    if continuity_dict["type"] == "Normal":
+        hist_dict = oflsfu.Read_ACIS_History_file(continuity_dict["cont_load_path"])
+    else:
+        # The review load is a TOO, SCS-107 or Full stop Return to Science load
+        # So read the review load's ACIS-History_edit.txt file
+        hist_dict = oflsfu.Read_ACIS_History_file(load_week_path, edit=True)
+
+    # Now set the value of previous and present instrument
+    # from the hist dictionary
+    state_dict["present_fp_instrument"] = hist_dict["instrume"]
+    state_dict["previous_fp_instrument"] = hist_dict["instrume"]
+
+    # Capture the RADMON status and store in the state dictionary
+    state_dict["radmon_state"] = hist_dict["radmon_status"]
+
+    # Find and read the CR*.backstop file
+    # First formulate the path to the CR*.backstop file with wild cards
+    partial_path = "/".join((load_week_path, "CR*.backstop"))
+
+    # Get the name of the CR*.backstop file. There is only one per OFLS directory
+    backstop_file_path = glob.glob(partial_path)[0]
+
+    # Create an instance of the Backstop File Processing Class
+    BFCI = bfp.Backstop_File_Class()
+    
+    # Read the backstop file and extract all commands of command type:
+    #    ACISPKT, SIMTRANS,  OORMPDS, OORMPEN
+
+    all_commands = BFCI.Read_BS_File(backstop_file_path) 
+
+    # Extract the SIM, ACISPKT and RADMON commands out of all of the commands
+    sim_acis_commands = BFCI.Extract_Type_and_TLMSID(["SIMTRANS", "ACISPKT", "OORMPDS", "OORMPEN"], all_commands)
+
+    # ----------------------------------  BEGIN DWELL PROCESSING ---------------------------------------------------
+    # The assumption here is that at the start of a load, you will not be in the middle of
```

Comment:
Maybe not critical, but it seems to me that if a dwell began before the end of the last load (e.g., if they did some maneuvering for whatever reason after the last observation at the tail end of the continuity load), this would be missed at the initialization here.